Description: Emit Upstart events when starting the login session and desktop session

Index: gdm-3.12.2/data/Init.in
===================================================================
--- gdm-3.12.2.orig/data/Init.in
+++ gdm-3.12.2/data/Init.in
@@ -6,6 +6,10 @@
 PATH="@X_PATH@:$PATH"
 OLD_IFS=$IFS
 
+if [ ! -d /run/systemd/system ]; then
+  /sbin/initctl -q emit login-session-start DISPLAY_MANAGER=gdm
+fi
+
 gdmwhich () {
   COMMAND="$1"
   OUTPUT=
Index: gdm-3.12.2/data/PreSession.in
===================================================================
--- gdm-3.12.2.orig/data/PreSession.in
+++ gdm-3.12.2/data/PreSession.in
@@ -7,3 +7,7 @@
 # Note that output goes into the .xsession-errors file for easy debugging
 #
 PATH="@X_PATH@:$PATH"
+
+if [ ! -d /run/systemd/system ]; then
+  /sbin/initctl -q emit desktop-session-start DISPLAY_MANAGER=gdm
+fi
